src/liballoc/heap.rs

```diff
@@ -345,47 +345,77 @@ mod imp {
           not(jemalloc),
           windows))]
 mod imp {
-    use libc::{c_void, size_t};
-    use libc;
+    use libc::{BOOL, DWORD, HANDLE, LPVOID, SIZE_T};
     use super::MIN_ALIGN;
 
-    extern {
-        fn _aligned_malloc(size: size_t, align: size_t) -> *mut c_void;
-        fn _aligned_realloc(block: *mut c_void, size: size_t,
-                            align: size_t) -> *mut c_void;
-        fn _aligned_free(ptr: *mut c_void);
+    extern "system" {
+        fn GetProcessHeap() -> HANDLE;
+        fn HeapAlloc(hHeap: HANDLE, dwFlags: DWORD, dwBytes: SIZE_T) -> LPVOID;
+        fn HeapReAlloc(hHeap: HANDLE, dwFlags: DWORD, lpMem: LPVOID, dwBytes: SIZE_T) -> LPVOID;
+        fn HeapFree(hHeap: HANDLE, dwFlags: DWORD, lpMem: LPVOID) -> BOOL;
+    }
+
+    #[repr(C)]
+    struct Header(*mut u8);
+
+    #[inline]
+    unsafe fn get_header<'a>(ptr: *mut u8) -> &'a mut Header {
+        &mut *(ptr as *mut Header).offset(-1)
+    }
+
+    #[inline]
+    unsafe fn align_ptr(ptr: *mut u8, align: usize) -> *mut u8 {
+        let aligned = ptr.offset((align - (ptr as usize & align - 1)) as isize);
```

Comment:
Could you add some parens to the `align - 1` just to make the binding explicit? (I can never remember the precedence of `-` vs `&`